pkg/report/writer.go

```diff
@@ -51,6 +52,19 @@ func Write(report types.Report, option Option) error {
 	var writer Writer
 	switch option.Format {
 	case FormatTable:
+		if option.Tree {
+			switch report.ArtifactType {
+			case ftypes.ArtifactFilesystem, ftypes.ArtifactRemoteRepository: // only `fs` and `repo` commands support `--dependency-tree` flag
```

Comment:
After consideration, I found we should disable `--dependency-tree` in the `image` subcommand. You can put nil as below.
https://github.com/aquasecurity/trivy/blob/ed322ac6664306c13ed673e0fa7dd6f76ebc9841/pkg/commands/app.go#L531-L533

Reply:
Thanks! i missed it. Now we don't need warning, because user will just get `unknown flag` error.